tools/code_coverage_report.bash

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
```

Comment:
im so excited to have code coverage reports!

Reply:
This is borrowed over from navigation2 and https://github.com/ros-planning/navigation2/pull/553 .
Perhaps @crdelsey would have some recommendations for your cmake setup?

Reply:
To have a single `COVERAGE_ENABLED` flag passed as a CMake argument, as @ruffsl has done in this PR, you need to add a few lines to every package's top level CMakelist.txt

```CMake
option(COVERAGE_ENABLED "Enable code coverage" FALSE)
  if(COVERAGE_ENABLED)
    add_compile_options(--coverage)
    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} --coverage")
    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} --coverage")
  endif()
```

Rather than duplicate this code everywhere, we created a package to hold common CMake macros.
https://github.com/ros-planning/navigation2/tree/master/nav2_common

This package defines a macro with our standard package header, and we just call that in each package. So instead of adding all the lines above, we add two lines to each package:

```CMake
find_package(nav2_common)  # for the macro definintion
```
and then after all the `find_package` lines

```CMake
nav2_package()  # to bring in standard package configuration options including COVERAGE_ENABLED
```

Reply:
@davetcoleman would you like to me to port something like `movit2_common` into this PR?

Reply:
That seems like a great optimization. Does it make sense to make a moveit2_common package or would it be better to rename the nav2_common package to something more general like ament_code_coverage and include that?